src/treetensornetworks/solvers/contract.jl

```diff
@@ -69,23 +70,37 @@ function contract(tn1::AbstractTTN, tn2::AbstractTTN; alg="fit", kwargs...)
   return contract(Algorithm(alg), tn1, tn2; kwargs...)
 end
 
-function contract(
-  tn1s::Vector{<:AbstractTTN}, tn2s::Vector{<:AbstractTTN}; alg="fit", kwargs...
-)
-  return contract(Algorithm(alg), tn1s, tn2s; kwargs...)
-end
-
 """
 Overload of `ITensors.apply`.
 """
-function apply(tn1::AbstractTTN, tn2::AbstractTTN; kwargs...)
-  tn12 = contract(tn1, tn2; kwargs...)
+function apply(tn1::AbstractTTN, tn2::AbstractTTN; init, kwargs...)
+  #plin=plev(first(externalinds(init)))
+  #outputindsnet=
+  #plout=plev(outputindsnet[first(vertices(outputindsnet))])
+  plev_inc = plev_diff(flatten_external_indsnetwork(tn1, tn2), external_indsnetwork(init))
+  init = prime(init, plev_inc)
+  tn12 = contract(tn1, tn2; init, kwargs...)
   return replaceprime(tn12, 1 => 0)
 end
 
-function apply(
-  tn1s::Vector{<:AbstractTTN}, tn2s::Vector{<:AbstractTTN}; alg="fit", kwargs...
+function sum_apply(
+  tns::Vector{<:Tuple{<:AbstractTTN,<:AbstractTTN}}; alg="fit", init, kwargs...
 )
-  tn12 = contract(tn1s, tn2s; alg="fit", kwargs...)
+  #plin=plev(first(externalinds(init)))
+  #outputindsnet = flatten_external_indsnetwork(first(tns), first(tn2s))
+  #plout=plev(outputindsnet[first(vertices(outputindsnet))])
+  plev_inc = plev_diff(
+    flatten_external_indsnetwork(first(first(tns)), last(first(tns))),
+    external_indsnetwork(init),
+  )
+  init = prime(init, plev_inc)
+  alg != "fit" && error("sum_apply not implemented for other algorithms than fit.")
```

Comment:
I think we can remove this check, and allow it to fail based on the function not being defined for the given `Algorithm` type.